zerver/lib/actions.py

```diff
@@ -3404,6 +3411,9 @@ def do_update_message_flags(user_profile: UserProfile,
 
     if operation == 'add':
         count = msgs.update(flags=F('flags').bitor(flagattr))
+        if flag == 'read':
+            do_increment_logging_stat(user_profile, COUNT_STATS['messages_read_log:client:day'],
+                                      None, timezone_now(), increment=count)
```

Comment:
I think it'd be cleaner to have this be calling a helper function of the form `log_number_read(user_profile, count)`